numpy/core/include/numpy/pythoncapi_compat.h

```diff
@@ -0,0 +1,342 @@
+// Header file providing new functions of the Python C API to old Python
+// versions.
+//
+// File distributed under the MIT license.
+//
+// Homepage:
+// https://github.com/pythoncapi/pythoncapi_compat
+//
+// Latest version:
+// https://raw.githubusercontent.com/pythoncapi/pythoncapi_compat/master/pythoncapi_compat.h
+//
+// SPDX-License-Identifier: MIT
+
+#ifndef PYTHONCAPI_COMPAT
+#define PYTHONCAPI_COMPAT
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <Python.h>
+#include "frameobject.h"          // PyFrameObject, PyFrame_GetBack()
```

Comment:
Why `"..."` for `frameobject.h`? It seems to be included in the python include directory in my distro.

Reply:
frameobject.h is not included by Python.h :-( I'm not sure why.

I partially fixed this issue in Python 3.10: Python.h now includes a new pyframe.h header file that I added, it defines PyFrameObject, PyFrame_GetLineNumber() and PyFrame_GetCode(), but not PyFrame_GetBack() (excluded from the limited C API).